src/connection-types/taxonomy/taxonomy.php

```diff
@@ -224,7 +224,14 @@ protected function create_term_for_object( $object_id ) {
 
 		wp_cache_set( 'o2o_term_exists_' . $this->taxonomy . '_' . $term_id, true );
 
-		add_post_meta( $object_id, 'o2o_term_id_' . $this->taxonomy, $term_id, true );
+		// add_post_meta( $object_id, 'o2o_term_id', $term_id, true );
+		// modified by Chris Wilson, 6/29/15
+		// perhaps due to imported content, we have a bunch of mismatched term-ids. This fixes it.
+
+		if (!add_post_meta( $object_id, 'o2o_term_id', $term_id, true)) {
+			update_post_meta ( $object_id, 'o2o_term_id', $term_id );
```

Comment:
Do we need to even check `add_post_meta()`? The `update_post_meta` function works both ways.